### PR TITLE
Feature/profile creation

### PR DESCRIPTION
**Pull request by Stefuu**
implementing buy alias feature

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: src/utils/types.ts [hunk 1]
```diff
@@ -88,3 +88,19 @@ export interface updateBucketProps {
 	objectCid: string
 	schemaDid: string
 }
+
```
Comment (tgfassina) [inline src/utils/types.ts:91]: @Stefuu  Is this being used? Couldn't find any

File: src/redux/slices/authenticationSlice.tsx [hunk 2]
```diff
@@ -86,6 +107,21 @@ export const authenticationSlice = createSlice({
 			state.error = true
 			state.loading = false
 		})
+
+		builder.addCase(userBuyAlias.pending, (state) => {
+			state.loading = true
+		})
+
+		builder.addCase(userBuyAlias.fulfilled, (state, action) => {
+			const { meta } = action
```
Comment (tgfassina) [inline src/redux/slices/authenticationSlice.tsx:116]: What's "meta"? This time we can't use `action.payload` like the others?

File: src/components/PrivateRoute/Container.tsx [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
 	Component: React.FC
 }
 
 function PrivateRoute({ Component, ...props }: PrivateRouteProps) {
 	const isLogged = useSelector(selectIsLogged)
+	const alias = useSelector(selectAlias)
 
-	return <>{isLogged ? <Component {...props} /> : <Navigate to="/" />}</>
+	return (
+		<>{isLogged && alias ? <Component {...props} /> : <Navigate to="/" />}</>
```
Comment (tgfassina) [inline src/components/PrivateRoute/Container.tsx:17]: I think here it should redirect to `/create-profile`.
Currently it puts me on the login screen (which is weird that we can see it even though we're logged in, but I don't think that's a big issue)

**Review by tgfassina — CHANGES_REQUESTED** [review]

File: src/redux/slices/authenticationSlice.tsx (hunk 2 above)
Comment (Stefuu) [reply src/redux/slices/authenticationSlice.tsx:116]: Payload is the response data, meta contains what we are sending on the request, in this case the response does not contain the alias, but our parameters do, so when the transaction is successful i`m just grabbing the alias we passed in and setting it to app state

File: src/utils/types.ts (hunk 1 above)
Comment (Stefuu) [reply src/utils/types.ts:91]: Its not, nice catch!

**Pull request closed without merging**